File: Selenium/Project/Activity3.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
# Create a WebDriver instance (assuming you have Firefox WebDriver installed)
driver = webdriver.Firefox()

# Navigate to the HRM system login page
driver.get("http://alchemy.hguy.co/orangehrm")

# Find the username and password input fields and login button
username_field = driver.find_element(By.ID,"txtUsername")
password_field = driver.find_element(By.ID,"txtPassword")
login_button = driver.find_element(By.ID,"btnLogin")

# Enter the login credentials
username_field.send_keys("orange")
password_field.send_keys("orangepassword123")

# Click the login button
login_button.click()

wait = WebDriverWait(driver, 10)
wait.until(EC.presence_of_element_located((By.ID, "menu_dashboard_index")))

# Close the browser
driver.quit()

File: Selenium/Project/Activity3.java
package HRM;

    import io.github.bonigarcia.wdm.WebDriverManager;
    import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.testng.Assert;
import org.testng.annotations.AfterTest;
import org.testng.annotations.BeforeTest;
import org.testng.annotations.Test;

public class Activity3 {

        WebDriver driver;

        @BeforeTest
        public void setUp() {
            WebDriverManager.chromedriver().setup();
            //Create a new instance of the Firefox driver
            driver = new ChromeDriver();
            // Open the URL
            driver.get("http://alchemy.hguy.co/orangehrm");
        }

        @Test
        public void loginToOrangeHRM() {
            // Find and fill the username and password fields using different selectors
            WebElement usernameField = driver.findElement(By.id("txtUsername")); // Using ID
            WebElement passwordField = driver.findElement(By.cssSelector("input#txtPassword")); // Using CSS Selector

            // Enter login credentials
            usernameField.sendKeys("orange");
            passwordField.sendKeys("orangepassword123");

            // Click on the Login button
            driver.findElement(By.id("btnLogin")).click();

            // Verify the homepage has opened
            String expectedTitle = "OrangeHRM"; // The title of the homepage
            String actualTitle = driver.getTitle();
            Assert.assertEquals(actualTitle, expectedTitle, "Homepage title doesn't match");

        }

        @AfterTest
        public void tearDown() {
            driver.quit();
        }
    }


